src/darksend.cpp

```diff
@@ -36,42 +36,39 @@ void CDarksendPool::ProcessMessage(CNode* pfrom, std::string& strCommand, CDataS
 
     if(strCommand == NetMsgType::DSACCEPT) {
 
-        PoolMessage nMessageID = MSG_NOERR;
-
         if(pfrom->nVersion < MIN_PRIVATESEND_PEER_PROTO_VERSION) {
-            nMessageID = ERR_VERSION;
             LogPrintf("DSACCEPT -- incompatible version! nVersion: %d\n", pfrom->nVersion);
-            PushStatus(pfrom, STATUS_REJECTED, nMessageID);
+            PushStatus(pfrom, STATUS_REJECTED, ERR_VERSION);
             return;
         }
 
         if(!fMasterNode) {
-            nMessageID = ERR_NOT_A_MN;
             LogPrintf("DSACCEPT -- not a Masternode!\n");
-            PushStatus(pfrom, STATUS_REJECTED, nMessageID);
+            PushStatus(pfrom, STATUS_REJECTED, ERR_NOT_A_MN);
             return;
         }
 
         int nDenom;
         CTransaction txCollateral;
         vRecv >> nDenom >> txCollateral;
 
+        LogPrint("privatesend", "DSACCEPT -- nDenom %d (%s)  txCollateral %s", nDenom, GetDenominationsToString(nDenom), txCollateral.ToString());
+
         CMasternode* pmn = mnodeman.Find(activeMasternode.vin);
         if(pmn == NULL) {
-            nMessageID = ERR_MN_LIST;
-            PushStatus(pfrom, STATUS_REJECTED, nMessageID);
+            PushStatus(pfrom, STATUS_REJECTED, ERR_MN_LIST);
             return;
         }
 
-        if(nSessionUsers == 0 && pmn->nLastDsq != 0 &&
+        if(vecSessionCollaterals.size() == 0 && pmn->nLastDsq != 0 &&
             pmn->nLastDsq + mnodeman.CountEnabled(MIN_PRIVATESEND_PEER_PROTO_VERSION)/5 > mnodeman.nDsqCount)
         {
             LogPrintf("DSACCEPT -- last dsq too recent, must wait: addr=%s\n", pfrom->addr.ToString());
-            nMessageID = ERR_RECENT;
-            PushStatus(pfrom, STATUS_REJECTED, nMessageID);
+            PushStatus(pfrom, STATUS_REJECTED, ERR_RECENT);
             return;
         }
 
+        PoolMessage nMessageID = MSG_NOERR;
```

Comment:
Hmm, not sure how.. We need `nMessageID` to be returned from `IsDenomCompatibleWithSession` (and later from both `CreateNewSession` or `AddUserToExistingSession` which I'm going to split `IsDenomCompatibleWithSession` to in another PR) to push it to clients. Otherwise we would need to move `PushStatus` inside of `IsDenomCompatibleWithSession` (etc) and I'd like to keep all these status stuffs in `ProcessMessage` rather that all over the place (all of current Push/RelayStatus in other parts of code will go away in following PRs too).